connectors/sources/box.py

```diff
@@ -176,6 +177,12 @@ async def _handle_client_errors(self, exception):
         skipped_exceptions=NotFound,
     )
     async def get(self, url, headers, params=None):
+        query_string = (
+            "&".join(f"{key}={value}" for key, value in params.items())
+            if params
+            else ""
+        )
+        self._logger.debug(f"Calling GET {url}?{query_string}")
```

Comment:
I think `query_string` should only built when debug logs are enabled. How it is here, this string is being built every run even though it isn't used unless debug logs are enabled. It's minor, but every bit counts when looking at performance.

This could be moved to a func elsewhere called something like `debug_query_string` and could be called in-line like this:

```suggestion
        self._logger.debug(f"Calling GET {url}?{debug_query_string(params)}")
```